### PR TITLE
return errors when it happened

### PR DESCRIPTION
**Pull request by vtolstov**
Signed-off-by: Vasiliy Tolstov <v.tolstov@unistack.org>

### CODE REVIEW

**Comment by softwarejl** [comment]
Can we use the V2 interface to compatible with the previous one?

**Comment by hzhou8** [comment]
Thanks @vtolstov ! I haven't looked every details yet but overall it looks good to me. I agree with @softwarejl on the compatibility concern. There are two ways to handle it. 1) As mentioned by @softwarejl , use V2 interface. 2) We may branch and release version 0.1 before merging this. Both approaches have pros and cons. For option 1), we will need to maintain 2 versions of interfaces for certain period. For option 2), old interface users won't get new features/bug fixes unless those are backported to v0.1 branch. Any thoughts on this?

**Comment by vtolstov** [comment]
what do you think about go.mod ?

**Comment by vtolstov** [comment]
I'm preferred option 2. So if you need the new features expected to have some breaks....

**Comment by vtolstov** [comment]
after that pr, I'm plan to push dhcp support, and after some fixes logical router support

**Comment by hzhou8** [comment]
@vtolstov thanks for the updates. I'll let @softwarejl do a detailed review. BTW, I saw some changes unrelated to what the patch was described to do "return errors when it happened", such as the changes for GetInstance(). If it is needed, it is better to be splitted as a separate commit. It is preferred that one commit fixes one problem.

For the compatibility problem, I think we agreed with option 2) branch and release version 0.1, and put this change as v0.2. I will do the branching.

File: goovn/ovndb.go [hunk 1]
```diff
@@ -75,31 +73,28 @@ type OVNDB struct {
 	imp *ovnDBImp
 }
 
-var once sync.Once
-var ovnDBApi OVNDBApi
```
Comment (softwarejl) [inline goovn/ovndb.go]: why we remove that?

Comment (softwarejl) [reply goovn/ovndb.go]: we must keep only one instance in one process.

File: goovn/uuid.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package goovn
+
+import (
+	"github.com/google/uuid"
+)
+
+func newUUID() (string, error) {
```
Comment (softwarejl) [inline goovn/uuid.go]: what is this for?

File: goovn/ovndb.go (hunk 1 above)
Comment (vtolstov) [reply goovn/ovndb.go]: Ah, ok.. I'm prefer to use client like semantics..
Does it strict requirement to have single instance?

File: goovn/uuid.go (hunk 2 above)
Comment (vtolstov) [reply goovn/uuid.go]: I'm change from rand to proper uuid, i think that is more generic and have less ability to duplicate..

**Comment by vtolstov** [comment]
Ok im revert methid name GetInstance

**Comment by vtolstov** [comment]
im return ol method name and once

**Comment by vtolstov** [comment]
Does i need to change someting else? (i'm plan to use this branch to logical router and dhcp)

**Comment by vtolstov** [comment]
Thanks

**Pull request merged**